lib/scholar/impute/knn_imputer.ex

```diff
@@ -0,0 +1,256 @@
+defmodule Scholar.Impute.KNNImputer do
+  @moduledoc """
+  Imputer for completing missing values using k-Nearest Neighbors.
+
+  Each sample's missing values are imputed using the mean value from
+    `n_neighbors` nearest neighbors found in the training set. Two samples are
+    close if the features that neither is missing are close.
+  """
+  import Nx.Defn
+  import Scholar.Metrics.Distance
+
+  @derive {Nx.Container, keep: [:missing_values], containers: [:statistics]}
+  defstruct [:statistics, :missing_values]
+
+  opts_schema = [
+    missing_values: [
+      type: {:or, [:float, :integer, {:in, [:nan]}]},
```

Comment:
Why `:nan` here is wrapped inside `:in`?

Reply:
could you suggest how to change it please?
because this is not compiling 
type: {:or, [:float, :integer,:nan]},

Reply:
This is correct as is. float and integer are types, nan is a value. :)